spec/requests/admin/tables_spec.rb

```diff
@@ -33,7 +33,7 @@ def app
     @headers = {
       'CONTENT_TYPE'  => 'application/json',
     }
-    host! "#{@user.username}.localhost.lan"
+    host! "localhost/user/#{@user.username}"
```

Comment:
The others are clear improvements since they now use the helper!

This one, I'm not so sure. There's a concept in carto called "subdomainless" that basically controls whether your urls are constructed with the user as a subdomain `<user>.localhost.lan` or as a part of the path `localhost.lan/user/<user>/`.

The helpers should be able to take care of that for you, and carto shouldn't really care if it's one or the other. Check:

config/app_config.yml:14

See if that is properly set.

Reply:
The config is properly set for subdomainless_urls.  That being said though only carto code honors this.  Therefore another way to solve this short of the hardcode that exists in the original and proposed change is to use carto's url apis such as:

`host! CartoDB.base_url(@user.username).sub!(/^https?\:\/\//, '')`